flow/connectors/kafka/kafka.go

```diff
@@ -242,38 +241,61 @@ Loop:
 				break Loop
 			}
 
-			ls.Push(fn)
-			ls.Push(pua.LuaRecord.New(ls, record))
-			err := ls.PCall(1, -1, nil)
-			if err != nil {
-				return nil, fmt.Errorf("script failed: %w", err)
-			}
-			args := ls.GetTop()
-			for i := range args {
-				kr, err := lvalueToKafkaRecord(ls, ls.Get(i-args))
+			var err error
+			chain, err = utils.LuaPoolJob(wgCtx, pool, chain, func(ls *lua.LState, wait <-chan struct{}) {
```

Comment:
ideally be a method on `pool` SubmitJob or something.

Reply:
pool is `puddle.Pool[*lua.LState]` so can't add a method, but once this is validated I'll look to move more code into utils for sharing with pubsub

Reply:
Replaced `puddle` with a finite number of coroutines pulling from message channels while pushing back to channels which a coordinator goroutine reads in proper order. Channel juggling now entirely hidden behind abstraction